### DISCUSSION TITLE
idea: warn on unbounded queue creation and uncollected asyncio task creation

### DISCUSSION THREAD

**Discussion by earonesty** [astral-sh/ruff#8387] [Ideas]
quite often a future bug to create an unbounded queue.   

guaranteed a bug to create an asyncio task when you don't collect the return value (auto cancels the task when it goes out of scope)

**Comment by charliermarsh** [comment]
We currently have [`asyncio-dangling-task`](https://docs.astral.sh/ruff/rules/asyncio-dangling-task/) which is maybe similar to what you're describing.

**Reply by zanieb** [reply]
Auto-cancels the task when it gets garbage collected sometime in the future :D I'm so glad we have a rule for that — it's a devious behavior. 

@earonesty Can you please include an example of the queue creation?

**Comment by earonesty** [comment]
x  = queue.Queue()

You can leave the max size empty

explicitly setting it to none would be a nice way of overriding the linter
behavior

basically when you have a que with no maximum size, there are only two
possibilities

1. You secretly know that you're never going to push more than a certain
amount onto it, or that the consumer is going to always move faster than
the producer, so you're not worried about it

2. You don't care if the program crashes when it runs out of ram

In the first case it's probably a reasonable to add a maximum size

In the second case it's reasonable to make the behavior explicit or provide
a linter override
